pkg/controller/ingress-shim/controller.go

```diff
@@ -72,41 +67,35 @@ type controller struct {
 // It returns the workqueue to be used to enqueue items, a list of
 // InformerSynced functions that must be synced, or an error.
 func (c *controller) Register(ctx *controllerpkg.Context) (workqueue.RateLimitingInterface, []cache.InformerSynced, error) {
-	// construct a new named logger to be reused throughout the controller
-	c.log = logf.FromContext(ctx.RootContext, ControllerName)
+	kShared := ctx.KubeSharedInformerFactory
+	cmShared := ctx.SharedInformerFactory
 
-	// create a queue used to queue up items to be processed
-	c.queue = workqueue.NewNamedRateLimitingQueue(controllerpkg.DefaultItemBasedRateLimiter(), ControllerName)
+	c.log = logf.FromContext(ctx.RootContext, ControllerName)
+	queue := workqueue.NewNamedRateLimitingQueue(controllerpkg.DefaultItemBasedRateLimiter(), ControllerName)
 
-	// obtain references to all the informers used by this controller
-	ingressInformer := ctx.KubeSharedInformerFactory.Networking().V1beta1().Ingresses()
-	certificatesInformer := ctx.SharedInformerFactory.Certmanager().V1().Certificates()
-	issuerInformer := ctx.SharedInformerFactory.Certmanager().V1().Issuers()
-	// build a list of InformerSynced functions that will be returned by the Register method.
-	// the controller will only begin processing items once all of these informers have synced.
 	mustSync := []cache.InformerSynced{
-		ingressInformer.Informer().HasSynced,
-		certificatesInformer.Informer().HasSynced,
-		issuerInformer.Informer().HasSynced,
+		cmShared.Certmanager().V1().Certificates().Informer().HasSynced,
```

Comment:
issue: This is duplicated, one of them should be in the ingress informer.
```suggestion
		kShared.Networking().V1beta1().Ingresses().Informer().HasSynced,
```

Reply:
Fixed in 1cb39d1efeb703a6d2c5f5b0e5e5c4ceeed2463a.

Reply:
This no longer syncs on ingresses, is that intentional?

Reply:
Oops, that's an good catch 😅 Somehow I read "Ingress" instead of "Issuer".

I wish we had a way to unit-test this, but I can't find a clean way without having to refactor a lot 😞

Reply:
The previous code seems to sync both ingresses and issuers, and now it only syncs one of them, is that intentional?

Reply:
This one change is intentional since Issuers (and ClusterIssuers) are not watched in this controller which means `Issuers().Informer().HasSynced` does not make sense.

Not sure why we were doing `HasSynced` on Issuers and ClusterIssuers in the old code though.

Apologies for the confusion I created with this change.

Reply:
Line 88 is still syncing clusterissuers, it sounds like we no longer want to do that?

Reply:
You are also correct, there should not be anything related to issuers and clusterissuers here. I thus removed three fields from the `controller` struct:

- `helper` (has never been used in this controller, probably a copy-paste leftover from the issuer controller)
- `issuerLister` (unused),
- `clusterIssuerLister` (unused)

It should be good now.